app/models.py

```diff
@@ -345,6 +402,38 @@ def supplier(cls):
     def framework(cls):
         return db.relationship(Framework, lazy='joined', innerjoin=True)
 
+    @declared_attr
+    def lot(cls):
+        return db.relationship(Lot, lazy='joined', innerjoin=True)
+
+    @validates('service_id')
+    def validate_service_id(self, key, value):
+        if not is_valid_service_id(value):
+            raise ValidationError("Invalid service ID value '{}'".format(value))
+
+        return value
+
+    @validates('status')
+    def validates_status(self, key, value):
+        if value not in self.STATUSES:
+            raise ValidationError("Invalid status value '{}'".format(value))
+
+        return value
+
+    @validates('data')
```

Comment:
I like this `@validates` decorator - nice.  But a validation method feels like an odd place to do all this data manipulation (dropping fields, stripping whitespace).  I'm not sure... it is nice and neat having it here though.